pyrefly/lib/test/tuple.rs

```diff
@@ -397,6 +397,14 @@ def test(x: tuple[int] | tuple[str]) -> None:
 "#,
 );
 
+testcase!(
+    test_tuple_concat_large_union,
+    r#"
+def test(a: tuple[int, ...] | bool) -> None:
```

Comment:
hmm it looks like even in dev mode this test passes before your changes. I wonder if you had the same experience? I'd like to make sure this regression test ensures we don't accidentally create this issue again.

Reply:
aa, I checked again, this test worked before, but origin issue's will trigger very long error.

I set a function becase use raw a will raise `a` is uninitialized.

will update this test example.